### PR TITLE
Tyt 8

### PR DESCRIPTION
**Pull request by JakMal9**
Buy pesticides

### CODE REVIEW

File: weltyt.py [hunk 1]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
 
         print("Invalid input, write integer only please. ")
 
+def buy_pesticides(pesticides):
+    amount = input("How many pesticides do you want to buy? Write only integer please. ")
```
Comment (SMaciej) [inline weltyt.py:48]: `provide` would be better instead of `write`.

File: weltyt.py [hunk 2]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
...
+
+    try:
+        amount = int(amount)
```
Comment (SMaciej) [inline weltyt.py:51]: We want to catch only this line, not the whole logic after it.

File: weltyt.py [hunk 3]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
...
+
+        money_amount = cur.execute(
```
Comment (SMaciej) [inline weltyt.py:53]: DRY https://en.wikipedia.org/wiki/Don%27t_repeat_yourself
We used exactly the same code before - move it to some method or at least assign the SQL to some kind of constant.

File: weltyt.py [hunk 4]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
...
+            'SELECT amount  FROM resources WHERE type = "Money";'
+        ).fetchone()[0]
+        pesticides_price = cur.execute(
+            'SELECT price FROM resources WHERE type = "Pesticides";'
+        ).fetchone()[0]
+        new_money_amount = money_amount - amount * pesticides_price
+        cur.execute(
```
Comment (SMaciej) [inline weltyt.py:60]: ditto

File: weltyt.py [hunk 5]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
...
+            f'UPDATE resources SET amount = {new_money_amount}  WHERE type = "Money";'
+        )
+        pesticides_amount = cur.execute(
```
Comment (SMaciej) [inline weltyt.py:63]: I would consider creating a generic function for this whole update money/resource logic, as DRY as possible.

File: weltyt.py [hunk 6]
```diff
@@ -44,6 +44,33 @@ def buy_plants(plants):
...
+            'SELECT amount  FROM resources WHERE type = "Pesticides";'
+        ).fetchone()[0]
+        new_pesticides_amount = pesticides_amount + amount
+        cur.execute(
+            f'UPDATE resources SET amount = {new_pesticides_amount}  WHERE type = "Pesticides";'
+        )
+        print("Well done pesticides deal. ")
+    except ValueError:
+
+        print("Invalid input, write integer only please. ")
```
Comment (SMaciej) [inline weltyt.py:73]: `provide`

**Review by SMaciej — CHANGES_REQUESTED** [review]

**Comment by SMaciej** [comment]
What about applying black?

File: weltyt.py (hunk 2 above)
Comment (JakMal9) [reply weltyt.py:51]: But we have to. Otherwise it doesn't work. We tried it before.

**Comment by JakMal9** [comment]
Black is ready, it was done in branch TYT-16.

File: weltyt.py [hunk 7]
```diff
@@ -15,33 +15,63 @@ def show_my_resources(x):
     for rows in cur.execute("SELECT * FROM lands"):
         print(rows)
 
+def get_resource(column, resource_type):
+        return cur.execute(
```
Comment (maciejsobolewski) [inline weltyt.py:19]: black

**Review by maciejsobolewski — APPROVED** [review]

**Pull request merged**